ui/overview_page.go

```diff
@@ -309,55 +271,185 @@ func (pg *overviewPage) centralize(gtx layout.Context, content layout.Widget) la
+	walletNameVisibility := len(c.info.Wallets) < 1
+	var transactionRows []layout.Widget
+	theme := pg.theme
+	for index, txn := range (*pg.walletTransactions).Recent {
+		txnWidgets := transactionWidgets{
+			wallet:  theme.Body1(txn.WalletName),
+			balance: txn.Balance,
+			date:    theme.Body1(txn.DateTime),
+			status:  theme.Body1(""),
+		}
+		if txn.Txn.Direction == dcrlibwallet.TxDirectionSent {
+			txnWidgets.direction = c.icons.sendIcon
+		} else {
+			txnWidgets.direction = c.icons.receiveIcon
+		}
+
+		if txn.Status == "confirmed" {
+			txnWidgets.status.Text = formatDateOrTime(txn.Txn.Timestamp)
+			txnWidgets.statusIcon = c.icons.confirmIcon
+		} else {
+			txnWidgets.status.Text = txn.Status
+			txnWidgets.statusIcon = c.icons.pendingIcon
+		}
+
+		txnWidgets.direction.Scale = 1
+		txnWidgets.statusIcon.Scale = 1
+
+		click := pg.toTransactionDetails[index]
+
+		row := func(gtx C) D {
+			pointer.Rect(image.Rectangle{Max: gtx.Constraints.Max}).Add(gtx.Ops)
+			click.Add(gtx.Ops)
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
 				layout.Rigid(func(gtx C) D {
-					return layout.Inset{Top: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
-						return txn.status.Layout(gtx)
+					heightWithResolution := pg.txnRowHeight * int(gtx.Metric.PxPerDp)
+					gtx.Constraints.Min.Y, gtx.Constraints.Max.Y = heightWithResolution, heightWithResolution
+					return layout.W.Layout(gtx, func(gtx C) D {
+						return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+							layout.Rigid(func(gtx C) D {
+								i := layout.Inset{
+									Right: values.MarginPadding16,
+								}
+								return i.Layout(gtx, func(gtx C) D {
+									return txnWidgets.direction.Layout(gtx)
+								})
+							}),
+							layout.Flexed(1, func(gtx C) D {
+								return layout.Flex{
+									Axis:      layout.Horizontal,
+									Alignment: layout.Middle,
+									Spacing:   layout.SpaceBetween,
+								}.Layout(gtx,
+									layout.Rigid(func(gtx C) D {
+										return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+											layout.Rigid(func(gtx C) D {
+												return layoutBalance(gtx, txnWidgets.balance, c)
+											}),
+											layout.Rigid(func(gtx C) D {
+												if !walletNameVisibility {
+													return layout.Dimensions{}
+												}
+												return layout.Inset{Bottom: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
+													return decredmaterial.Card{
+														Color: pg.theme.Color.Background,
+													}.Layout(gtx, func(gtx C) D {
+														return layout.Inset{
+															Top:   values.MarginPadding2,
+															Left:  values.MarginPadding5,
+															Right: values.MarginPadding5,
+														}.Layout(gtx, func(gtx C) D {
+															txnWidgets.wallet.TextSize = values.TextSize12
+															txnWidgets.wallet.Color = pg.theme.Color.Gray
+															return txnWidgets.wallet.Layout(gtx)
+														})
+													})
```

Comment:
This transaction row can be broken down into smaller methods or functions to make it easier to visualize the layout of the UI with a quick glance at the code. This for instance can be kept in a method with a name "walletBadge".